### PR TITLE
Analytics Client

### PR DESCRIPTION
**Pull request by mhayes853**
Implementation of an analytics dependency. It can be used like such in components/hooks:

```tsx
const Component = () => {
    const analytics = useDependencyValue(analyticsDependencyKey)

    const doThing = () => {
        analytics("thingHappened", { attribute: "hello", metric: 5 })
    }

    return <Button label="Do Thing" onPress={doThing} />
}
```

### CODE REVIEW

File: tests/AmplifyAnalytics.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+describe("AmplifyAnalytics tests", () => {
+  it("converts an event with no data payload correctly", () => {
+    amplifyRecordAnalytics(testAnalyticEventName)
+    expect(recordedEvent).toMatchObject({ name: testAnalyticEventName })
+  })
+
+  it("converts an event with only numeric data to amplify analytics metrics", () => {
+    const eventData = { testMetric: testAnalyticMetric }
+    amplifyRecordAnalytics(testAnalyticEventName, eventData)
```
Comment (seanim0920) [inline tests/AmplifyAnalytics.test.ts:25]: 👍 Easy format to follow

File: lib/Analytics.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  eventName: string,
+  data?: Record<string, string | number>
+) => {
+  Analytics.record({
+    name: eventName,
+    attributes: compactMapAnaylticsData(data ?? {}, (value) => {
+      if (typeof value === "string") return value
+      return null
+    }),
+    metrics: compactMapAnaylticsData(data ?? {}, (value) => {
```
Comment (seanim0920) [inline lib/Analytics.ts:34]: 👍

File: amplify/backend/types/amplify-dependent-resources-ref.d.ts [hunk 3]
```diff
@@ -1,81 +1,96 @@
 export type AmplifyDependentResourcesAttributes = {
-    "auth": {
```
Comment (seanim0920) [inline amplify/backend/types/amplify-dependent-resources-ref.d.ts:2]: Looks like whitespace changes, I'll try adding an eslint rule to fix this

File: amplify/backend/function/postNotifications/function-parameters.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
       "resourceName": "backendResources",
       "version": "Always choose latest version",
       "isLatestVersionSelected": true,
-      "env": "backuptwo"
```
Comment (seanim0920) [inline amplify/backend/function/postNotifications/function-parameters.json:26]: Thanks for fixing this

File: amplify/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Getting Started with Amplify CLI
```
Comment (seanim0920) [inline amplify/README.md:1]: I don't think me or Tristan got this README when we were using amplify, our versions may be outdated

**Review by seanim0920 — APPROVED** [review]

**Review by nicoletteL — APPROVED** [review]

**Review by suryatiru — APPROVED** [review]

**Pull request merged**